src/tests/Regressions/coreclr/GitHub_49826/test49826.cs

```diff
@@ -11,6 +11,10 @@ class Program
     static int Main()
     {
         JsonSerializerOptions options = new JsonSerializerOptions();
+
+        // This is needed to internally call RootBuiltInConvertersAndTypeInfoCreator
+        JsonSerializer.Serialize("a-string");
```

Comment:
The breaking change here was actually caused and discussed in this PR: https://github.com/dotnet/runtime/pull/51300#discussion_r614917792.

cc @steveharter, @eiriktsarpalis, @eerhardt 

The STJ source generator actually doesn't generate calls to `JsonSerializerOptions.GetConverter(Type)`, so we don't need to break users here. We could invoke the [converter-relevant logic](https://github.com/dotnet/runtime/blob/1cc11b1fadfd508b2d906e1f2706e1835b9cf014/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs#L30-L45) in the `RootBuiltInConvertersAndTypeInfoCreator` method to root all the built-in converters. However, this would mean that a single call to `GetConverter` in would root all the converters and make the size bigger in a non-pay-for-play way. If we prevent a breaking change here, then this PR can be closed. Otherwise, it would be good to add another test variant here that takes the approach described in https://github.com/dotnet/runtime/pull/51300#discussion_r615496306.

Reply:
Can we decide ASAP what to do here, and possibly disable this test until whatever is decided is implemented?

Reply:
Looking at what this test is trying to test - https://github.com/dotnet/runtime/issues/49826 - the original issue trying to be tested is when the converters are trying to be initialized early in a static ctor.

Is adding `JsonSerializer.Serialize("a-string");` here invalidating the original intent of the test?

Reply:
Well, according to my understanding of your analysis of the issue and subsequent David's fix the problem was caused by internal inlining / devirtualization within GetDefaultSimpleConverters and that method hasn't changed at all, the only difference is that now it's called from <code>RootBuiltInConvertersAndTypeInfoCreator</code>, not from the cctor. In other words, I don't think that the refactoring automatically invalidates the test even though I do admit I haven't tested that locally by rolling back David's fix from March.

For @layomia's comment, please note that the purpose of the test is not to actually test serializer functionality; it's just that we noticed that the exact code sequence in <code>GetDefaultSimpleConverters</code> happened to trip a previously unseen codegen issue in Crossgen2 context. With this in mind I don't see why we'd need to test multiple ways to activate the serializers, we just need GetDefaultSimpleConverters to run once in the test.

Reply:
I chatted offline with @steveharter / @eiriktsarpalis. We determined to maintain backwards-compat for this scenario - https://github.com/dotnet/runtime/pull/51897. After that PR is merged, I believe that this one can be closed.

Reply:
@layomia Presumably the test would also need to be re-enabled (i.e., https://github.com/dotnet/runtime/pull/51759 reverted)?

Reply:
@BruceForstall yes I'll send a PR.